From 83bcdfcef8c46df1b49a8b493dd1c99020e7f7ad Mon Sep 17 00:00:00 2001
From: P Dheeraj Srujan Kumar <p.dheeraj.srujan.kumar@intel.com>
Date: Mon, 20 Feb 2023 19:12:17 +0530
Subject: [PATCH] Fix Build issue downstream

The upstream build fails when static_cast is called out, whereas the
downstream build expects the static_cast.
This patch applies the required static_cast to fix the build issue.

Signed-off-by: P Dheeraj Srujan Kumar <p.dheeraj.srujan.kumar@intel.com>
---
 hsbp-manager/src/hsbp_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hsbp_manager.cpp b/src/hsbp_manager.cpp
index bc92a32..ed05b12 100644
--- a/src/hsbp_manager.cpp
+++ b/src/hsbp_manager.cpp
@@ -2578,7 +2578,7 @@ void loadHsbpConfig()
                     /* Clear and Load all details to global hsbp configuration
                      * variable */
                     hsbpConfig.clearConfig();
-                    hsbpConfig.rootBus = *rootI2cBus;
+                    hsbpConfig.rootBus = static_cast<size_t>(*rootI2cBus);
                     hsbpConfig.supportedHsbps = std::move(*supportedHsbps);
 
                     for (auto& clkBuffType : *clockBufferTypes)
-- 
2.34.1

